src/builder/commands/composer.rs

```diff
@@ -0,0 +1,252 @@
+    cmd.arg("--no-interaction");
+    Ok(cmd)
+}
+
+pub fn composer_install(distro: &mut Box<Distribution>, ctx: &mut Context,
+    pkgs: &Vec<String>)
+    -> Result<(), String>
+{
+    let features = scan_features(&ctx.composer_settings);
+    try!(packages::ensure_packages(distro, ctx, &features));
+
+    if pkgs.len() == 0 {
+        return Ok(());
+    }
+
+    let mut cmd = try!(composer_cmd(ctx));
+    cmd.args(&["global", "require", "--prefer-dist", "--update-no-dev"]);
+    cmd.args(pkgs);
+    try!(run(cmd));
+    Ok(())
+}
+
+pub fn composer_dependencies(distro: &mut Box<Distribution>,
+    ctx: &mut Context, info: &ComposerDepInfo)
+    -> Result<(), StepError>
+{
+    let features = scan_features(&ctx.composer_settings);
+    try!(packages::ensure_packages(distro, ctx, &features));
+
+    let mut cmd = try!(composer_cmd(ctx));
+    cmd.arg("install");
+    if let Some(ref dir) = info.working_dir {
+        cmd.arg(format!("--working-dir={}", dir));
+    }
+    if !info.dev { cmd.arg("--no-dev"); }
+    if info.ignore_platform_reqs { cmd.arg("--ignore-platform-reqs"); }
+    if info.no_autoloader { cmd.arg("--no_autoloader"); }
+    if info.no_scripts { cmd.arg("--no-scripts"); }
+    if info.no_plugins { cmd.arg("--no-plugins"); }
+    if info.optimize_autoloader { cmd.arg("--optimize-autoloader"); }
+
+    match info.prefer {
+        Some(ref p) if p == "dist" => { cmd.arg("--prefer-dist"); },
+        Some(ref p) if p == "source" => { cmd.arg("--prefer-source"); },
+        Some(ref p) => return Err(From::from(format!(
+            "Value of 'ComposerDependencies.prefer' must be either \
+            'source' or 'dist', '{}' given", p
+        ))),
+        _ => {}
+    }
+
+    run(cmd)
+}
+
+pub fn configure(ctx: &mut Context) -> Result<(), String> {
+    try!(ctx.add_cache_dir(Path::new("/tmp/composer-cache"),
+                           "composer-cache".to_string()));
+
+    ctx.environ.insert("COMPOSER_HOME".to_owned(),
+                       COMPOSER_HOME.to_owned());
+    ctx.environ.insert("COMPOSER_VENDOR_DIR".to_owned(),
+                       COMPOSER_VENDOR_DIR.to_owned());
+    ctx.environ.insert("COMPOSER_BIN_DIR".to_owned(),
+                       COMPOSER_BIN_DIR.to_owned());
+    ctx.environ.insert("COMPOSER_CACHE_DIR".to_owned(),
+                       COMPOSER_CACHE.to_owned());
+
+    Ok(())
+}
+
+pub fn bootstrap(ctx: &mut Context) -> Result<(), String> {
+    try_msg!(create_dir(COMPOSER_HOME, true),
+        "Error creating composer home dir {d:?}: {err}", d=COMPOSER_HOME);
+
+    let composer_inst = try!(download::download_file(ctx, COMPOSER_BOOTSTRAP));
+    try!(copy(&composer_inst, &Path::new("/vagga/root/tmp/composer-setup.php"))
+        .map_err(|e| format!("Error copying composer installer: {}", e)));
+
+    let runtime_exe = ctx.composer_settings
+        .runtime_exe
+        .clone()
+        .unwrap_or(DEFAULT_RUNTIME.to_owned());
+
+    let args = [
+        "/usr/bin/env".to_owned(),
```

Comment:
In my tests ubuntu didn't find `php` command, it has to `php5` so I thought it would be safer to use `/usr/bin/env`. I'll take a look at that again to see if it is really needed